app/com/linkedin/drelephant/tuning/Flow.java

```diff
@@ -0,0 +1,112 @@
+package com.linkedin.drelephant.tuning;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.log4j.Logger;
+
+
+/**
+ *  This is the Flow , which create four pipeline and each pipeline executes in seperate thread.
+ *  Four pipeline are
+ *  BaseLineManager : This will have two manager BaselineManagerOBT, BaselineManagerHBT
+ *  JobStatusManagerPipeline: This will have one manager , AzkabanJobStatusManager
+ *  FitnessManager : This will have three managers FitnessManagerHBT,FitnessManagerOBTAlgoIPSO,FitnessManagerOBTPSO
+ *  TuningTypeManager : This will have Combinations of TuningType , AlgorithmType and execution engine. for e.g
+ *  TuningTypeManagerOBTAlgoIPSO for Map Reduce and Spark ...
+ */
+public class Flow {
+  List<List<Manager>> pipelines = null;
+  private static final Logger logger = Logger.getLogger(Flow.class);
+
+  public Flow() {
+    pipelines = new ArrayList<List<Manager>>();
+    createBaseLineManagersPipeline();
+    createJobStatusManagersPipeline();
+    createFitnessManagersPipeline();
+    createTuningTypeManagersPipeline();
+  }
+
+  public List<List<Manager>> getPipeline() {
+    return pipelines;
+  }
+
+  public void executeFlow() throws InterruptedException {
+    for (final List<Manager> pipelineType : this.pipelines) {
+      Thread t1 = new Thread(new Runnable() {
+        @Override
+        public void run() {
+          for (Manager manager : pipelineType) {
```

Comment:
currently one thread is getting created for Baseline, one thread for JobStatus , one for fitness and one for tuningtype. Are u suggesting to have one thread for each manager , one thread for
BaselineMgrOBT, and another one for BaselineMgrHBT. We can do it ,but then it will create many threads and each thread would be very light weight. May be down the line ,once we have many jobs , then we can move to approach ,which you are being suggested